src/main/java/shop/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,68 @@
+package shop.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import shop.dto.request.shopping.cart.AddBookToCartRequestDto;
+import shop.dto.request.shopping.cart.UpdateQuantityBookRequestDto;
+import shop.dto.responce.cart.item.CartItemResponseDto;
+import shop.dto.responce.shopping.cart.ShoppingCartResponseDto;
+import shop.service.ShoppingCartService;
+
+@RestController
+@RequestMapping(value = "/cart", produces = MediaType.APPLICATION_JSON_VALUE)
+@RequiredArgsConstructor
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
```

Comment:
Create separate method to get `Long userId` from `Authentication authentication` and provide it to each Service method
`User user = (User) authentication.getPrincipal();`